core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -466,7 +465,9 @@ class DefaultMessageFormatter extends MessageFormatter {
   var keyDeserializer: Option[Deserializer[_]] = None
   var valueDeserializer: Option[Deserializer[_]] = None
 
-  override def init(props: Properties): Unit = {
+  override def configure(configs: Map[String, _]): Unit = {
+    val props = new java.util.Properties()
+    configs.asScala.foreach { case (key, value) => props.put(key, value.toString) }
```

Comment:
Can't we use `configs` directly instead of building `props`?

Reply:
I initially was doing this but this forces casting the value to strings everywhere due to the type of the Map here `[String, _]`. Keeping the `Properties` object turned out way simpler.

Reply:
That's a good point. Probably, the correct way to do this in the future would be to define a `ConfigDef` to parse the Map. We can tackle this as part of https://github.com/apache/kafka/pull/8909.